apps/pkeyutl.c

```diff
@@ -309,10 +314,15 @@ int pkeyutl_main(int argc, char **argv)
     }
 
     pkey = get_pkey(kdfalg, inkey, keyform, key_type, passinarg, pkey_op, e);
+
+    if (pkey_op == EVP_PKEY_OP_VERIFYRECOVER && is_EdDSA(pkey)) {
```

Comment:
Wouldn't it be more logical to do a positive check instead of a negative check? 
I.e. check for algos that do support verifyrecover (i.e. RSA), instead of reject it just for EdDSA. I guess there are more algos than just EdDSA that do NOT support verifyrecover (ECDSA also does not support verifyrecover AFIK, same for ML-DSA, etc). 

Something like this:
`if (pkey_op == EVP_PKEY_OP_VERIFYRECOVER && !is_RSA(pkey)) {`

Reply:
Good point - done.